packages/perseus/src/widgets/plotter/plotter.tsx

```diff
@@ -1080,18 +1073,30 @@ export class Plotter extends React.Component<Props, State> implements Widget {
     };
 
     setPicHeight: (arg1: number, arg2: number) => void = (i, y) => {
-        const values = [...this.state.values];
+        const values = [...this.props.userInput];
         values[i] = y;
-        this.drawPicHeights(values, this.state.values);
-        this.setState({values: values});
-        this.changeAndTrack({values: values});
+        this.drawPicHeights(values, this.props.userInput);
+        this.changeAndTrack(values);
     };
 
-    changeAndTrack: (arg1: any) => void = (data) => {
-        this.props.onChange(data);
+    changeAndTrack: (userInput: PerseusPlotterUserInput) => void = (
+        userInput,
+    ) => {
+        this.props.handleUserInput(userInput);
         this.props.trackInteraction();
     };
 
+    /**
+     * Plotter uses Graphie and Graphie is inaccessible,
+     * this helps us populate user input in tests
+     * (not great, but not a lot of options)
+     *
+     * @internal
+     */
+    _testInsertUserInput(userInput: PerseusPlotterUserInput) {
```

Comment:
Ooo interesting, I was wondering where this was coming from. 

I think this makes sense for now, but I wonder if we shouldn't consider making a ticket to find a more holistic / standardized / better way to handle this. It seems like a possible idiosyncrasy that we'd wonder about in the future. 

Maybe it's simply not worthwhile if our goal is to move away from Graphie anyway.

Reply:
Plotter is a hidden widget, so I think it's fine to ignore the problem until it can be replaced with IG.